src/rr/drawfuns/R_DrawFuzzColumn.java

```diff
@@ -1,275 +1,342 @@
             }
 
-            public void invoke() {
-                int count;
-                int dest;
+            // Does not work with blocky mode.
+            dest = computeScreenDest();
+
+            // Looks like an attempt at dithering,
+            // using the colormap #6 (of 0-31, a bit
+            // brighter than average).
+            if (count > 4) {// MAES: unroll by 4
+                do {
+                    // Lookup framebuffer, and retrieve
+                    // a pixel that is either one column
+                    // left or right of the current one.
+                    // Add index from colormap to index.
+
+                    screen[dest] = blurryTable.computePixel(screen[dest + fuzzoffset[fuzzpos]]);
+
+                    // Clamp table lookup index.
+                    if (++fuzzpos == FUZZTABLE)
+                        fuzzpos = 0;
 
-                // Adjust borders. Low...
-                if (dcvars.dc_yl == 0)
-                    dcvars.dc_yl = 1;
+                    dest += SCREENWIDTH;				
 
-                // .. and high.
-                if (dcvars.dc_yh == dcvars.viewheight - 1)
-                    dcvars.dc_yh = dcvars.viewheight - 2;
+                    screen[dest] = blurryTable.computePixel(screen[dest + fuzzoffset[fuzzpos]]);
+                    if (++fuzzpos == FUZZTABLE) fuzzpos = 0;
+                    dest += SCREENWIDTH;
 
-                count = dcvars.dc_yh - dcvars.dc_yl;
+                    screen[dest] = blurryTable.computePixel(screen[dest + fuzzoffset[fuzzpos]]);
+                    if (++fuzzpos == FUZZTABLE) fuzzpos = 0;
+                    dest += SCREENWIDTH;
 
-                // Zero length.
-                if (count < 0)
-                    return;
+                    screen[dest] = blurryTable.computePixel(screen[dest + fuzzoffset[fuzzpos]]);
+                    if (++fuzzpos == FUZZTABLE) fuzzpos = 0;
+                    dest += SCREENWIDTH;
+
+                } while ((count-=4) > 4);
+
+                if (count > 0) {
+                    do {
+                        screen[dest] = blurryTable.computePixel(screen[dest + fuzzoffset[fuzzpos]]);
+
+                        // Clamp table lookup index.
+                        if (++fuzzpos == FUZZTABLE)
+                            fuzzpos = 0;
 
-                if (RANGECHECK) {
-                        super.performRangeCheck();
+                        dest += SCREENWIDTH;
+                    } while (count-- > 0);
                 }
+            }
+        }
+    }
+
+    public static final class TrueColor extends R_DrawFuzzColumn<byte[], int[]> {
+
+        public TrueColor(int SCREENWIDTH, int SCREENHEIGHT, int[] ylookup,
+                int[] columnofs, ColVars<byte[], int[]> dcvars,
+                int[] screen, IDoomSystem I, BlurryTable BLURRY_MAP) {
+            super(SCREENWIDTH, SCREENHEIGHT, ylookup, columnofs, dcvars, screen, I, BLURRY_MAP);
+        }
+
+        @Override
```

Comment:
We must be able to activate Maes's blurry effect. This is unique to Mocha Doom and must be preserved, though it must be turned off by default in the config file. By default, the fuzz effect is used to remain as close as possible to vanilla Doom.